### PR TITLE
Allow administrator to specify the user for engine admin

### PR DESCRIPTION
**Pull request by turboFei**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala [hunk 1]
```diff
@@ -196,7 +196,11 @@ private[v1] class AdminResource extends ApiRequestContext with Logging {
       @QueryParam("sharelevel") shareLevel: String,
       @QueryParam("subdomain") subdomain: String,
       @QueryParam("hive.server2.proxy.user") hs2ProxyUser: String): Response = {
-    val userName = fe.getSessionUser(hs2ProxyUser)
+    val userName = if (isAdministrator(fe.getRealUser())) {
+      Option(hs2ProxyUser).getOrElse(fe.getRealUser())
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala:200]: it is tricky but I do not want to add a new param for this.

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala:200]: cc @pan3793

**Comment by pan3793** [comment]
The basic idea is that administrator can proxy any user even it's not allowed by `core-site.xml`?

**Comment by turboFei** [comment]
> The basic idea is that administrator can proxy any user even it's not allowed by core-site.xml?

yes.

**Review by pan3793 — APPROVED** [review]
LGTM, we can centralize all logic about proxy user in the future.

**Comment by turboFei** [comment]
thanks, merging to master

**Pull request closed without merging**